src/coreclr/tools/Common/Compiler/Logger.cs

```diff
@@ -63,6 +70,14 @@ public void LogWarning(string text, int code, TypeSystemEntity origin, string su
                 Writer.WriteLine(warning.Value.ToMSBuildString());
         }
 
+        public void LogWarning(TypeSystemEntity origin, DiagnosticId id, params string[] args)
+        {
+            MessageOrigin messageOrigin = new MessageOrigin(origin);
+            MessageContainer? warning = MessageContainer.CreateWarningMessage(this, messageOrigin, id, args);
+            if (warning.HasValue)
+                Writer.WriteLine(warning.Value.ToMSBuildString());
+        }
```

Comment:
Can be simplified to:
```suggestion
        public void LogWarning(TypeSystemEntity origin, DiagnosticId id, params string[] args) =>
            LogWarning(new MessageOrigin(origin), id, args);
```